brave/brave-core
# Anti-Targeting for Brave Ads

Resolves https://github.com/brave/brave-browser/issues/14224

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Make sure the anti-targeting resource `mkdhnfmjhklfnamlheoliekgeohamoig`  is added to the staging component for the respective country (matching the test machine system's settings) and the components `models.json` has been updated.
- Make sure to have a test catalog with some anti-targeting test lists that match a creative set id from the catalog that is being triggered during test (see below for example)
- Without having visited a site on the anti-targeting list, verify that the ad with the respective ID is being delivered
- Now visit a site on the list and verify that the ad with the respective ID is *not* being delivered `[87942:775:0401/172513.733280:VERBOSE2:exclusion_rule_util.h(26)] creativeSetId 5bdeab83-048f-48a7-9602-a1092ded123c excluded as user has previously visited an anti-targeting site`

Example anti-targeting test lists
```
{
  "locale": "gb",
  "version": 1,
  "timestamp": "2021-03-32 00:00:00",
  "sites": {
    "5bdeab83-048f-48a7-9602-a1092ded123c": [
      "https://www.brave.com",
      "https://www.bravesoftware.com/"
    ],
    "ccdaf616-b8a8-481a-aa8b-caf5fb131353": [
      "https://www.bar.org"
    ]
  }
}
```
